### ISSUE TITLE
AbstractInterner equalBytes() returning true when strings are not equal.

### ISSUE THREAD

**Issue by chrisjeg** [OpenHFT/Chronicle-Bytes#84]
There appears to be an issue in the string interning implementation (AbstractInterner). The following unit tests demonstrates a case where invoking the intern method can return a different value to that passed in. This seems to be caused by equalBytes returning true when strings are not equal.

This has been reproduced in 1.16.23 and also the most recent version of chronicle-bytes. 

```
@Test
public void internFailingTest() throws IORuntimeException {
    UTF8StringInterner utf8StringInterner = new UTF8StringInterner(4096);

    utf8StringInterner.intern(Bytes.from("DELEGATE: 912796UF4 Notional is > limit 25.00mm"));
    utf8StringInterner.intern(Bytes.from("TW-TRSY-20181217-NY572677_3256N1"));
    String intern = utf8StringInterner.intern(Bytes.from("TW-TRSY-20181217-NY572677_3256N15"));
    assertThat(intern, equalTo("TW-TRSY-20181217-NY572677_3256N15"));
}

```
```
@Test
public void equalBytesFailingTest() throws IORuntimeException {
    BytesStore store1 = Bytes.from("TW-TRSY-20181217-NY572677_3256N1");
    BytesStore store2 = Bytes.from("TW-TRSY-20181217-NY572677_3256N15");
    assertThat(store1.equalBytes(store2, 33), equalTo(false));
}
```

**Comment by MartyIX** [comment]
@chrisjeg `equalBytes` is implemented as an asymmetric comparison. 

The commit above seems to fix the issue. However, it's big change and I don't really know what this may break. So ...

**Comment by JerryShea** [comment]
Have added some additional testing to capture another failure case and fixed the defect. Please review!

**Comment by JerryShea** [comment]
Thanks guys for drawing this one to our attention

**Comment by MartyIX** [comment]
LGTM

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.17.14](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.17.14), [BOM-2.17.196](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.196)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.17.14](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.17.14), [BOM-2.17.225](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.225)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.17.14](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.17.14), [BOM-2.17.241](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.241)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.20.101](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
